test/Dialect/Moore/canonicalizers.mlir

```diff
@@ -192,6 +192,19 @@ func.func @StructExtractFold2(%arg0: !moore.i17, %arg1: !moore.i42) -> (!moore.i
   return %1, %2 : !moore.i17, !moore.i42
 }
 
+// CHECK-LABEL: func.func @structExtractRefLower2Inject
+func.func @structExtractRefLower2Inject(%arg0: !moore.ref<struct<{a: i32, b: i32}>>) -> (!moore.ref<struct<{a: i32, b: i32}>>) {
+  // CHECK-NEXT: [[C42:%.+]] = moore.constant 42
+  // CHECK-NEXT: [[TMP1:%.+]] = moore.read %arg0 : <struct<{a: i32, b: i32}>>
+  // CHECK-NEXT: [[TMP2:%.+]] = moore.struct_inject [[TMP1]], "a", [[C42]] : struct<{a: i32, b: i32}>, i32
+  // CHECK-NEXT: moore.blocking_assign %arg0, [[TMP2]] : struct<{a: i32, b: i32}>
+  // CHECK-NEXT: return %arg0 : !moore.ref<struct<{a: i32, b: i32}>>
+  %0 = moore.constant 42 : i32
+  %1 = moore.struct_extract_ref %arg0, "a" : <struct<{a: i32, b: i32}>> -> <i32>
+  moore.blocking_assign %1, %0 : i32
```

Comment:
It's perfect if you can add some other cases. Such as:
```
%0 = moore.constant 42 : i32
%1 = moore.struct_extract_ref %arg0, "a" : <struct<{a: i32, b: i32}>> -> <i32>
moore.blocking_assign %1, %0 : i32

%2 = moore.constant 43 : i32 
moore.blocking_assign %1, %2 : i32
```
And
```
%0 = moore.constant 42 : i32
%1 = moore.struct_extract_ref %arg0, "a" : <struct<{a: i32, b: i32}>> -> <i32>
moore.blocking_assign %1, %0 : i32

%2 = moore.constant 43 : i32 
%3 = moore.struct_extract_ref %arg0, "b" : <struct<{a: i32, b: i32}>> -> <i32>
moore.blocking_assign %2, %3 : i32
```

Reply:
These tests may be complicated I can print the result of your case here
```
%0 = moore.constant 43 : i32
    %1 = moore.constant 42 : i32
    %2 = moore.read %arg0 : <struct<{a: i32, b: i32}>>
    %3 = moore.struct_inject %2, "a", %1 : struct<{a: i32, b: i32}>, i32
    moore.blocking_assign %arg0, %3 : struct<{a: i32, b: i32}>
    %4 = moore.read %arg0 : <struct<{a: i32, b: i32}>>
    %5 = moore.struct_inject %4, "a", %0 : struct<{a: i32, b: i32}>, i32
    moore.blocking_assign %arg0, %5 : struct<{a: i32, b: i32}>
    %6 = moore.read %arg0 : <struct<{a: i32, b: i32}>>
    %7 = moore.struct_inject %6, "a", %0 : struct<{a: i32, b: i32}>, i32
    moore.blocking_assign %arg0, %7 : struct<{a: i32, b: i32}>
    return %arg0 : !moore.ref<struct<{a: i32, b: i32}>>
```
and
```
%0 = moore.constant 43 : i32
    %1 = moore.constant 42 : i32
    %2 = moore.read %arg0 : <struct<{a: i32, b: i32}>>
    %3 = moore.struct_inject %2, "a", %1 : struct<{a: i32, b: i32}>, i32
    moore.blocking_assign %arg0, %3 : struct<{a: i32, b: i32}>
    %4 = moore.read %arg0 : <struct<{a: i32, b: i32}>>
    %5 = moore.struct_inject %4, "b", %0 : struct<{a: i32, b: i32}>, i32
    moore.blocking_assign %arg0, %5 : struct<{a: i32, b: i32}>
    return %arg0 : !moore.ref<struct<{a: i32, b: i32}>>
```